common/src/chain/transaction/transaction_index/mod.rs

```diff
@@ -26,6 +26,8 @@ pub enum Spender {
     RegularInput(Id<Transaction>),
     #[codec(index = 1)]
     StakeKernel(Id<Block>),
+    #[codec(index = 2)]
+    Premine(Id<Genesis>),
```

Comment:
This is an example of what we've discussed. There's no reason to distinguish between genesis and other kinds of blocks at the output level.

Reply:
Adding a new variant here seemed like a path of least resistance at the time. I don't see much harm in how it's done currently, but I think it could be changed to just a variant for `Transaction` and a variant for `GenBlock` with some effort.

Reply:
Ok, I introduced just one variant with `Id<GenBlock>` and it just works.